portal-impl/src/com/liferay/portal/util/PortalImpl.java

```diff
@@ -3508,7 +3508,14 @@ else if (groupId > 0) {
 					locale = LocaleUtil.fromLanguageId(defaultLanguageId);
 
 					if (LanguageUtil.isAvailableLocale(groupId, locale)) {
-						if (initialize) {
+						HttpServletRequest originalHttpServletRequest =
+							getOriginalServletRequest(httpServletRequest);
+
+						String currentURL =
+							(String)originalHttpServletRequest.getAttribute(
```

Comment:
@chileces @locpham97 Is this always non-null?

Reply:
@dantewang This can be null. I will update this source code and moreover I think this can be added a test

Reply:
Hi @dantewang, because the step to reproduce this issue passed a multiple filter and request. Therefore, the writing test is really complicated and I am not able to write it.

Reply:
@locpham97 Did you try to write an integration test or unit test? I think at least a unit test to cover this exact logic can be possible.

Reply:
@locpham97 You can just call `PortalImpl.getLocale` with certain input that simulates what you've seen in your investigation, and provide an expected output.

Reply:
I will try follow your instruction today. Thanks

Reply:
I just update the test for this issue. Please take time to review this ticket.